### ISSUE TITLE
Upgrade to nullsafe dependencies

### ISSUE THREAD

**Issue by b-jan** [zino-app/graphql-flutter#823]
**Describe the issue**
Many libraries are now using http `^0.13.0`
But `graphql` and `graphql_flutter` libs are still using http `^0.12.1`

It makes them incompatible with many libraries.

**To Reproduce**
1. Install `firebase_auth: ^1.0.0`
2. Install `graphql: ^4.0.1`
3. flutter pub get
4. See incompatibility error

**Expected behavior**
`graphql_flutter` and `graphql` libs should use http: 0.13.0 as a dependency

**Incompatibility Error**
On firebase_auth example:

```
[my_app] flutter pub get
Running "flutter pub get" in my_app...                        
Because firebase_auth >=1.0.0 depends on firebase_auth_web ^1.0.0 which depends on http_parser ^4.0.0, firebase_auth >=1.0.0 requires http_parser ^4.0.0.

And because graphql >=4.0.0-alpha.0 depends on http ^0.12.1 which depends on http_parser >=0.0.1 <4.0.0, firebase_auth >=1.0.0 is incompatible with graphql >=4.0.0-alpha.0.
So, because my_app depends on both firebase_auth ^1.0.0 and graphql ^4.0.1, version solving failed.

pub get failed (1; So, because my_app depends on both firebase_auth ^1.0.0 and graphql ^4.0.1, version solving failed.)
```

**Comment by TolgaYld** [comment]
same issue....

```
Because no versions of google_fonts match >2.0.0 <3.0.0 and google_fonts 2.0.0 depends on path_provider ^2.0.0, google_fonts ^2.0.0 requires path_provider ^2.0.0.
And because graphql_flutter >=4.0.0-alpha.0 depends on path_provider ^1.6.10, google_fonts ^2.0.0 is incompatible with graphql_flutter >=4.0.0-alpha.0.
So, because "app" depends on both google_fonts ^2.0.0 and graphql_flutter ^4.0.1, version solving failed.
pub get failed (1; So, because "app" depends on both google_fonts ^2.0.0 and graphql_flutter ^4.0.1, version solving failed.)
```

**Comment by micimize** [comment]
if someone gets to this before I do, bump the `rxdart` version also

**Comment by b-jan** [comment]
I tried the upgrade but got the same problem with `normalize` lib...

```
normalize >=0.4.5 depends on gql ^0.12.4 and graphql depends on gql ^0.13.0-nullsafety.1, normalize >=0.4.5 is forbidden.
```

**Comment by micimize** [comment]
@b-jan I've published `4.0.3-alpha.1` that I think should allow you to use other null safe packages, and pushed the changes to your [version_update](https://github.com/b-jan/graphql-flutter/tree/version_update) branch. Please let me know how it goes.

@TolgaYld your font issue is different, and should be fixed in #818
